route/access_rules.go

```diff
@@ -50,6 +45,11 @@ func (t *Target) AccessDeniedHTTP(r *http.Request) bool {
 		}
 	}
 
```

Comment:
Moving this block doesn't do what you think.  You have orphaned the actual source IP parsing above making this function do the same thing as the XFF check block.